sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -641,6 +641,10 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
         Row(key, value, key + 1)
       }.toSeq)
     assert(df.schema.map(_.name) === Seq("key", "valueRenamed", "newCol"))
+
+    // Renaming to a column that contains "." character
+    val df2 = testData.toDF().withColumnRenamed("value", "value.Renamed")
+    assert(df2.schema.map(_.name) === Seq("key", "value.Renamed"))
```

Comment:
Please add more test cases that columns whose name has '.' can be accessed without backticks